### PR TITLE
[Feature] Expressions example plugin

### PR DESCRIPTION
**Pull request by ashwin-pc**
### Description
We have a few example plugins that demonstrate how to use the various OpenSearch Dashboards features like UIActions, Embeddables, etc. This add to that list by creating an example plugin for `expressions`. It has 3 basic example for the main 3 features for `embeddables`, a playground to explore expressions and finally an explorer to explore and discover all the registered expression functions

https://user-images.githubusercontent.com/20453492/162554356-6dda3c11-6ae6-4119-b693-ce254a094eae.mp4
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
    - [ ] `yarn test:jest`
    - [ ] `yarn test:jest_integration`
    - [ ] `yarn test:ftr`
- [ ] New functionality has been documented.
- [x] Commits are signed per the DCO using --signoff

### CODE REVIEW

File: examples/expressions_example/.i18nrc.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
+  "prefix": "expressionsExample",
+  "paths": {
+    "expressionsExample": "."
+  },
+  "translations": ["translations/ja-JP.json"]
```
Comment (kavilla) [inline examples/expressions_example/.i18nrc.json:6]: nit: probably can remove this. i know this scaffolded on generation

**Review by kavilla — DISMISSED** [review]
Awesome!! TY so much. Given that it is an example and not an end user functionality I think most follow-ups can be handle subsequently.

In the developer docs to run and access examples, or specifically to access this example?

File: examples/expressions_example/.i18nrc.json (hunk 1 above)
Comment (ashwin-pc) [reply examples/expressions_example/.i18nrc.json:6]: I wasn't sure if we were going to use these files for the localization that we are reintroducing. Figured if we were, its easier to start from this boilerplate, and if not we can just nuke this and all other `.i18nrc.json` files

**Comment by ashwin-pc** [comment]
@kavilla I'll add to the expression plugin's readme in a followup PR because i dont have all the details i want to add in it yet. As for how to run example plugins, we already have these instructions in the [Example plugin folder's readme](https://github.com/opensearch-project/OpenSearch-Dashboards/blob/1.x/examples/README.asciidoc). I will update this readme too in that followup PR.

**Comment by ashwin-pc** [comment]
While the initial PR only made changes to the examples directory, this revision makes 3 changes to the expressions plugin itself. 2 of those changes are w.r.t `ReactExpressionRenderer` which is current not used anywhere application and safe to include for 2.0. The other change is to uiState. This was a bug and prevented the uiState from initializing correctly when used outside the context of Visualize. It should also not be an issue for 2.0, but i can remove that if necessary so that this is included in the release.

**Review by kavilla — APPROVED** [review]

File: src/plugins/expressions/public/_expression_renderer.scss [hunk 2]
```diff
@@ -8,8 +8,13 @@
...
 
 .expExpressionRenderer__expression {
+  display: flex;
+  flex-direction: column;
   width: 100%;
   height: 100%;
+  overflow: auto;
+  position: relative;
+  flex: 1 1 100%;
 }
```
Comment (zengyan-amazon) [inline src/plugins/expressions/public/_expression_renderer.scss:18]: Will this scss change impact the style of UI components in other places?

**Review by zengyan-amazon — APPROVED** [review]

File: examples/expressions_example/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+module.exports = {
+  root: true,
+  extends: ['@elastic/eslint-config-kibana', 'plugin:@elastic/eui/recommended'],
+  rules: {
+    '@osd/eslint/require-license-header': 'off',
```
[the remark this answers is not in the document]
Comment (ashwin-pc) [reply examples/expressions_example/.eslintrc.js:10]: None of the example files currently have license headers and our linter does not catch them either. Not sure why the examples were skipped, but i can add them to this and the other examples in a followup PR once i confirm the root cause.

File: examples/expressions_example/opensearch_dashboards.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+  "id": "expressionsExample",
+  "version": "1.0.0",
+  "opensearchDashboardsVersion": "opensearchDashboards",
+  "server": false,
+  "ui": true,
+  "requiredPlugins": [
+    "navigation",
+    "expressions",
+    "developerExamples",
+    "opensearchDashboardsReact"
```
[the remark this answers is not in the document]
Comment (ashwin-pc) [reply examples/expressions_example/opensearch_dashboards.json:11]: I didn't understand your comment @zengyan-amazon. These plugins are used in the expressions-example plugin. Hence marked as required dependencies.

File: src/plugins/expressions/public/_expression_renderer.scss (hunk 2 above)
Comment (ashwin-pc) [reply src/plugins/expressions/public/_expression_renderer.scss:18]: No they will not. These styles only affect `ReactExpressionRenderer` which is currently not used anywhere in the application.

**Pull request merged**